### ISSUE TITLE
Add filter keep_keys

### ISSUE THREAD

**Issue by vbotka** [ansible-collections/community.general#8438]
### Summary

Create filter keep_keys.

The collection ansible.utils provides the filter [keep_keys](https://github.com/ansible-collections/ansible.utils/blob/main/plugins/filter/keep_keys.py). Unfortunately:

* It is not working properly. See https://github.com/ansible-collections/ansible.utils/issues/353
* The documentation is poor. For example, it is not clear what "recursively" means in "Keep specific keys from a data recursively"
* There are only two examples and two tests. An example of "matching_parameter=regex" is missing. It is unclear whether the parameter *target* should be a list for *regex*.  A list of regex(es) would be an antipattern.
* But, the main hurdle to fix this filter is the [class AnsibleArgSpecValidator](https://github.com/ansible-collections/ansible.utils/blob/main/plugins/module_utils/common/argspec_validate.py) that is used in all ansible.utils/plugins. Such complexity is inefficient in a filter where a couple of comprehensions would do the job. This also violates the golden Unix rule [Tools, not policy](https://en.wikipedia.org/wiki/Unix_philosophy)

The goal is to keep the code simple and test it properly.

### Issue Type

Feature Idea

### Component Name

community.genera.keep_keys

### Additional Information

```yaml
  t: [k0, k1]
  l1:
    - {k0: A, k1: B, k2: C}
    - {k0: D, k1: E, k2: F}
  r1: "{{ l1 | community.general.keep_keys(target=t) }}"
```
gives

```yaml
  r1:
  - k0: A
    k1: B
  - k0: D
    k1: E
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

**Comment by vbotka** [comment]
For the record:  PR Add filter keep_keys  https://github.com/ansible-collections/community.general/pull/8439

**Comment by vbotka** [comment]
This feature request is implemented in PR #8456
